jl4-core/src/L4/EvaluateLazy/Machine.hs

```diff
@@ -1101,13 +1147,35 @@ initialEnvironment = do
   nilRef   <- AllocateValue ValNil
   evalContractRef <- AllocateValue =<< evalContractVal
   eventCRef <- AllocateValue eventCVal
+  plusRef <- AllocateValue $ ValBuiltinFun PlusFn
+  minusRef <- AllocateValue $ ValBuiltinFun MinusFn
+  timesRef <- AllocateValue $ ValBuiltinFun TimesFn
+  divideRef <- AllocateValue $ ValBuiltinFun DivideFn
+  moduloRef <- AllocateValue $ ValBuiltinFun ModuloFn
   isIntegerRef <- AllocateValue (ValUnaryBuiltinFun UnaryIsInteger)
   roundRef <- AllocateValue (ValUnaryBuiltinFun UnaryRound)
   ceilingRef <- AllocateValue (ValUnaryBuiltinFun UnaryCeiling)
   floorRef <- AllocateValue (ValUnaryBuiltinFun UnaryFloor)
   fulfilRef <- AllocateValue ValFulfilled
+
+  comparisonRefs <-
+    traverse
+      (\(funVal, uniq) -> do
+        r <- AllocateValue funVal
+        pure (uniq, r)
+      )
+      [ (ValBuiltinFun val, unique)
+      | (val, uniques) <-
+          [ (LtFun, TypeCheck.ltUniques)
+          , (LeqFun, TypeCheck.leqUniques)
+          , (GtFun, TypeCheck.gtUniques)
+          , (GeqFun, TypeCheck.geqUniques)
+          ]
+      , unique <- uniques
+      ]
+
```

Comment:
don't you just want to map the `Fun` over the uniques, then flatten ones and then allocate all? Seems to be a bit complicated for what it is... 

Perhaps you could also split up the traverse and the list.